### PR TITLE
Add support to allow pre-release opam

### PR DESCRIPTION
**Pull request by smorimoto**
Closes: https://github.com/ocaml/setup-ocaml/issues/698

### CODE REVIEW

File: packages/setup-ocaml/src/cache.ts [hunk 1]
```diff
@@ -71,13 +71,11 @@ async function composeOpamCacheKeys() {
...
-  const octokit = github.getOctokit(GITHUB_TOKEN);
-  const {
-    data: { tag_name: opamVersion },
-  } = await octokit.rest.repos.getLatestRelease({
-    owner: "ocaml",
-    repo: "opam",
-  });
+  const opamVersion =
+    platform === Platform.Win32
+      ? "0.0.0.2"
```
Comment (jchavarri) [inline packages/setup-ocaml/src/cache.ts:76]: Is this intended?

**Review by jchavarri — COMMENTED** [review]
Thanks!

Afaiu, this should be used like this?

```yaml
      - name: Use OCaml ${{ matrix.ocaml-compiler }}
        uses: ocaml/setup-ocaml@v2
        with:
          ocaml-compiler: ${{ matrix.ocaml-compiler }}
          allow-prelease-opam: true
```

**Comment by smorimoto** [comment]
> Afaiu, this should be used like this?

Yes 🙂

File: packages/setup-ocaml/src/cache.ts (hunk 1 above)
Comment (smorimoto) [reply packages/setup-ocaml/src/cache.ts:76]: This is as intended: on Windows, we continue to use the custom-built version of opam. Native support for Windows will be implemented after this PR is merged.

File: packages/setup-ocaml/src/opam.ts [hunk 2]
```diff
@@ -27,8 +28,8 @@ import {
 } from "./system";
 import { getCygwinVersion } from "./win32";
 
-async function getLatestOpamRelease() {
-  const semverRange = "<2.2.0";
+export async function getLatestOpamRelease() {
+  const semverRange = ALLOW_PRELEASE_OPAM ? "<2.3.0" : "<2.2.0";
```
Comment (smorimoto) [inline packages/setup-ocaml/src/opam.ts:32]: We could use `*` here instead of `<2.3.0` 🤷‍♂️

Comment (jchavarri) [reply packages/setup-ocaml/src/opam.ts:32]: Could be. I don't think it hurts to keep `2.3.0` though, whenever a new version is released, both numbers should be updated but they are really close to each other in the code, so it'd hard to miss the update to the prerelease version.

**Pull request merged**